napari/_qt/dialogs/_tests/test_screenshot_dialog.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+from qtpy.QtCore import QTimer
+from qtpy.QtWidgets import QApplication, QFileDialog, QLineEdit, QMessageBox
+
+from napari._qt.dialogs.screenshot_dialog import ScreenshotDialog
+from napari.utils.history import get_save_history
+
+
+@pytest.mark.parametrize("filename", ["test", "test.png", "test.tif"])
+def test_screenshot_save(qtbot, tmp_path, filename):
+    """Check passing different extensions with the filename."""
+
+    def save_function(path):
+        # check incoming path has extension event when a filename without one
+        # was provided
+        assert filename in path
+        if "." not in filename:
+            assert ".png" in path
+
+        # create a file with the given path to check for
+        # non-native qt overwrite message
+        with open(path, "w") as mock_img:
+            mock_img.write("")
+
+    qt_overwrite_shown = False
+
+    def handle_qt_overwrite_message():
+        for widget in QApplication.topLevelWidgets():
+            if isinstance(widget, QMessageBox):
+                # test should not enter here!
+                widget.accept()
+                nonlocal qt_overwrite_shown
+                qt_overwrite_shown = True
+                break
+
+    # setup dialog
+    dialog = ScreenshotDialog(
+        save_function, directory=str(tmp_path), history=get_save_history()
+    )
+    qtbot.addWidget(dialog)
+    dialog.setOptions(QFileDialog.DontUseNativeDialog)
+    dialog.show()
+
+    # check dialog and set filename
+    assert dialog.windowTitle() == 'Save screenshot'
+    line_edit = dialog.findChild(QLineEdit)
+    line_edit.setText(filename)
+
+    # check that no warning message related with overwriting is shown
+    QTimer.singleShot(100, handle_qt_overwrite_message)
+    dialog.accept()
+    qtbot.wait(100)
```

Comment:
```suggestion
    qtbot.wait(120)
```
For me it should be strictly greater than `singleShot`